tests/simulator_test.cpp

```diff
@@ -37,22 +37,23 @@ class EnergyCollectorTest : public ::testing::Test {
   enum class HitResult { ENERGY_COLLECTORS_EMPTY, NO_HIT, HIT };
 
 protected:
-  std::vector<std::unique_ptr<objects::EnergyCollector>> energyCollectors;
   MockModel nonEmptyModel, emptyModel;
 
   // performs ray hit at energy Collectors by modifying hitData.
   // Returns:
   // ENERGY_COLLECTORS_EMPTY - when no energyCollectors where assigned
   // NO_HIT - when there was no hit
   // HIT - when hit occurred;
-  [[nodiscard]] HitResult performHitCollector(const Ray &ray, float frequency,
-                                              RayHitData *hitData) {
+  [[nodiscard]] HitResult performHitCollector(
```

Comment:
Adding it here as I cannot comment on the commit message: passing parameters to functions is not *dependency injection*. Is a technique for managing dependencies in object oriented design, not defining params in functions.